json_object.c

```diff
@@ -675,9 +690,9 @@ int32_t json_object_get_int(const struct json_object *jso)
 }
 
 int json_object_set_int(struct json_object *jso,int new_value){
-	if (!jso || jso->o_type!=json_type_int)
+	if (!jso || jso->o_type!=json_type_int || jso->o.c_int.cint_type != json_object_int_type_int64)
```

Comment:
er.. only allowing int64 types through here is going to make for some non-obvious behavior.
Calling this should just switch cint_type to int64.

Reply:
ok, done